src/input/json_to_storage.hh

```diff
@@ -104,46 +225,140 @@ public:
      * Check if current head node is a JSON Object containing one key REF of type string.
      * If yes, returns the string through reference @p ref_address.
      */
-    bool get_ref_from_head(string & ref_address);
+    bool get_ref_from_head(string & ref_address) override;
+
+    // These methods are derived from PathBase
+    bool is_null_type() const override;
+    bool get_bool_value() const override;
+    std::int64_t get_int_value() const override;
+    double get_double_value() const override;
+    std::string get_string_value() const override;
+    std::string get_node_type() const override;
+    bool get_record_key_set(std::set<std::string> &) const override;
+    int get_array_size() const override;
+    bool is_record_type() const override;
+    bool is_array_type() const override;
+    PathJSON * clone() const override;
+    bool has_descendent_index(bool value_at_declaration) override;
+
+    inline std::string input_format_name() const override {
+    	return "JSON";
+    }
+
+    inline std::string sequence_name() const override {
```

Comment:
virtual method can be inlined. Moreover this way we do not guarantee that string returned by this method mach type name returned by get_node_type. Solution:
1) in constructor PathYAML, PathJSON make array of type name strings with common index for:
     - NULL
     - JSON object, YAML map
     - JSON array, YAML sequence
     some positions may be empty (e.g. Scalar for JSON)
2) Implement these methods in PathBase (reading particular index from the type name array)
3) in get_node_type use case to get particular items of the array.